includes/Modules/Analytics/Advanced_Tracking/Measurement_Events/WooCommerce_Event_List.php

```diff
@@ -18,134 +22,346 @@
+		foreach ( $category_ids as $category_id ) {
+			$category_query = get_term_by( 'id', $category_id, 'product_cat' );
+			if ( 'WP_Term' == get_class( $category_query ) ) {
+				$category_string .= $category_query->name . '/';
+			}
+		}
+		$item['category'] = substr( $category_string, 0, strlen( $category_string ) - 1 );
+		$item['id']       = $product->get_sku();
+		$item['name']     = $product_name;
+		$item['price']    = $product->get_price();
+		$item['quantity'] = 1;
+
+		$add_to_cart_meta                   = array();
+		$add_to_cart_meta['event_category'] = 'ecommerce';
+		$add_to_cart_meta['value']          = $product->get_price();
+		$add_to_cart_meta['currency']       = get_woocommerce_currency();
+		$add_to_cart_items                  = array();
+		$add_to_cart_items[]                = $item;
+		$add_to_cart_meta['items']          = $add_to_cart_items;
+		$add_to_cart_event                  = new Measurement_Event(
 			array(
 				'pluginName' => 'WooCommerce',
-				'category'   => 'ecommerce',
 				'action'     => 'add_to_cart',
-				'selector'   => '.woocommerce-page .single_add_to_cart_button',
+				'selector'   => '.woocommerce-page .add_to_cart_button[data-product_id="' . $product_id . '"]',
 				'on'         => 'click',
+				'metadata'   => $add_to_cart_meta,
 			)
 		);
-		$this->add_event( $event );
+		$this->add_event( $add_to_cart_event );
 
-		$event = new Measurement_Event(
-			array(
-				'pluginName' => 'WooCommerce',
-				'category'   => 'ecommerce',
-				'action'     => 'remove_from_cart',
-				'selector'   => '.woocommerce-page .remove',
-				'on'         => 'click',
-			)
-		);
-		$this->add_event( $event );
+	}
 
-		$event = new Measurement_Event(
-			array(
-				'pluginName' => 'WooCommerce',
-				'category'   => 'ecommerce',
-				'action'     => 'begin_checkout',
-				'selector'   => 'div.wc-proceed-to-checkout .checkout-button',
-				'on'         => 'click',
-			)
-		);
-		$this->add_event( $event );
+	/**
+	 * Creates relevant Measurement_Event objects when a single WooCommerce item is rendered.
+	 *
+	 * @since n.e.x.t.
+	 *
+	 * @param WC_Product $product The WooCommerce product that is being rendered.
+	 * @throws \Exception Thrown when invalid keys or value type.
+	 */
+	private function collect_wc_single_item( $product ) {
+		$product_name    = $product->get_name();
+		$item            = array();
+		$category_string = '';
+		$category_ids    = $product->get_category_ids();
+		foreach ( $category_ids as $category_id ) {
+			$category_query = get_term_by( 'id', $category_id, 'product_cat' );
+			if ( 'WP_Term' == get_class( $category_query ) ) {
+				$category_string .= $category_query->name . '/';
+			}
+		}
+		$item['category'] = substr( $category_string, 0, strlen( $category_string ) - 1 );
+		$item['id']       = $product->get_sku();
+		$item['name']     = $product_name;
+		$item['price']    = $product->get_price();
+		// TODO: Get the quantity of the item from the frontend.
```

Comment:
Let's still set an initial value for `quantity` here, so that this at least works. For example we'll have to accept getting the right quantity here won't be possible for AMP anyway if it's changed in the frontend. So let's specify the initial value here (which I'd assume is `1`?).